nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,361 @@
+
+        self.logger.debug(f"doughnut_status_df: {doughnut_status_df}")
+        status_df = pd.concat([status_df, doughnut_status_df], axis=1)
+        return status_df
+
+    def _check_bagel(self, status_df: pd.DataFrame) -> pd.DataFrame:
+        """Check the imaging bagel file (if exists)."""
+        nipoppy_checkpoint = "in_imaging_bagel"
+
+        self.logger.info(f"Status at nipoppy_checkpoint: {nipoppy_checkpoint}")
+
+        try:
+            bagel = self.bagel
+        except FileNotFoundError:
+            self.logger.warning(
+                f"No bagel file found at {self.layout.fpath_imaging_bagel}."
+            )
+            return
+
+        # Get the number of participants in the bagel
+        participant_ids = bagel[bagel.col_participant_id].unique()
+        session_ids = bagel[bagel.col_session_id].unique()
+        pipelines = bagel[bagel.col_pipeline_name].unique()
+
+        self.logger.info(f"\tNumber of participants in bagel: {len(participant_ids)}")
+        self.logger.info(f"\tAvailable visits (n={len(session_ids)}): {session_ids}")
+        self.logger.info(f"\tAvailable pipelines (n={len(pipelines)}): {pipelines}")
+
+        bagel[self.col_pipeline] = (
+            bagel[bagel.col_pipeline_name]
+            + "\n"
+            + ""
+            + bagel[bagel.col_pipeline_version]
+            + "\n"
+            + bagel[bagel.col_pipeline_step]
+        )
+
+        bagel_pipeline_df = bagel[
+            [
+                bagel.col_participant_id,
+                bagel.col_session_id,
+                self.col_pipeline,
+                bagel.col_status,
+            ]
+        ]
+        bagel_pipeline_df = bagel_pipeline_df[
+            bagel_pipeline_df[bagel.col_status] == STATUS_SUCCESS
+        ]
+
+        bagel_pipeline_df = bagel_pipeline_df.pivot(
+            index=[bagel.col_participant_id, bagel.col_session_id],
+            columns=self.col_pipeline,
+            values=bagel.col_status,
+        )
+
+        bagel_status_df = bagel_pipeline_df.groupby([bagel.col_session_id]).count()
+
+        self.logger.debug(f"bagel_status_df: {bagel_status_df}")
+
+        status_df = pd.concat([status_df, bagel_status_df], axis=1)
+        return status_df
+
+    def _add_rewards(self, status_df: pd.DataFrame) -> pd.DataFrame:
+        """Add rewards column to the status DataFrame."""
+        status_df = status_df.fillna(0).astype(int).reset_index()
+        status_df = status_df.sort_values(by=self.manifest.col_session_id)
+
+        # generate positive reinforcement emojis
+        clipboard = "📋"
+
+        sad_cat_emoji = "😿"
+        broom_emoji = "🧹"
+        doughnut_emoji = "🍩"
+        sparkles_emoji = "✨"
+        rocket_emoji = "🚀"
+        bagel_emoji = "🥯"
+        party_popper_emoji = "🎉"
+        confetti_emoji = "🎊"
+
+        # based on quantity
+        high_throughput = 100
+        bento_box_emoji = "🍱"  # 100 subjects in BIDS
+        mate_emoji = "🧉"  # 100 subjects processed
```

Comment:
I think it is confusing to have secret emojis 😅

Reply:
Also right now we have e.g. the broom emoji and the bento box emoji simultaneously -- are they supposed to be either/or or cumulative?

Reply:
You don't like secret restaurant menus? 
they were supposed to be cumulative, but happy to think to avoid too many emojis.

Reply:
What if we have the emojis next to the numbers inside the cells?

For example something like this:
```
| session_id | in_manifest | in_pre_reorg | in_post_reorg | in_bids | fmriprep | mriqc |
|------------|-------------|--------------|---------------|---------|----------|-------|
| 1          | 100 📋      | 100 🍩       | 100 🍩        | 100 ✨  | 100 🎉   | 100 🎉|
| 2          | 90 📋       | 80 🍩        | 80 🍩         | 70 🧹   | 60 🏃    | 70 🏃 |
```

We would be losing the bagel emoji and the 🎊 emoji though. Also a legend might not be necessary anymore? But not sure